extending.rst

```diff
@@ -0,0 +1,12 @@
+.. _extending:
+
+Extending Conan
+---------------
+
+This section provides an introduction to extension capabilities of Conan:
+
+.. toctree::
+   :maxdepth: 2
+
+   extending/python_requires
```

Comment:
Interesing section. I think I'd also move here (not in this PR) the "custom generators" section. Maybe also a good place to document the python API in the future.